spec/lib/rbac/filterer_spec.rb

```diff
@@ -765,29 +765,73 @@ def combine_filtered_ids(user_filtered_ids, belongsto_filtered_ids, managed_filt
           let(:admin_user) { FactoryGirl.create(:user, :role => "super_administrator") }
           let!(:cloud_template_root) { FactoryGirl.create(:template_cloud, :publicly_available => false) }
 
-          it 'returns all cloud templates when user is admin' do
-            results = described_class.filtered(TemplateCloud, :user => admin_user)
-            expect(results).to match_array(TemplateCloud.all)
+          let(:tenant_2) { FactoryGirl.create(:tenant, :parent => default_tenant, :source_type => 'CloudTenant') } # T2
+          let(:group_2) { FactoryGirl.create(:miq_group, :tenant => tenant_2) } # T1
+          let(:user_2) { FactoryGirl.create(:user, :miq_groups => [group_2]) }
+
+          context "when tenant doesn't have source_id" do
+            it 'returns all cloud templates when user is admin' do
+              User.current_user = admin_user
+              results = described_class.filtered(TemplateCloud, :user => admin_user)
+              expect(results).to match_array(TemplateCloud.all)
+            end
+
+            context "when user is restricted user" do
+              let(:tenant_3) { FactoryGirl.create(:tenant, :parent => tenant_2) } # T3
+              let!(:cloud_template) { FactoryGirl.create(:template_cloud, :tenant => tenant_3, :publicly_available => true) }
+
+              it "returns all public cloud templates" do
+                User.current_user = user_2
+                results = described_class.filtered(TemplateCloud, :user => user_2)
+                expect(results).to match_array([cloud_template, cloud_template_root])
+              end
+
+              context "should ignore" do
```

Comment:
please write reason why it should ignore instead of, is it because of publicly_available => false ?